### PR TITLE
[rampup] Stop txn/network after slot feature

### PR DESCRIPTION
**Pull request by joaosreis**
Merge #14516 into `rampup`.

### CODE REVIEW

**Comment by joaosreis** [comment]
!ci-build-me

**Comment by joaosreis** [comment]
!ci-build-me

File: src/lib/ledger_catchup/normal_catchup.ml [hunk 1]
```diff
@@ -79,7 +80,7 @@ let verify_transition ~context:(module Context : CONTEXT) ~trust_system
     Transition_handler.Validator.validate_transition
       ~context:(module Context)
       ~frontier ~unprocessed_transition_cache
-      enveloped_initially_validated_transition
+      enveloped_initially_validated_transition ~slot_tx_end ~slot_chain_end
```
Comment (joaosreis) [inline src/lib/ledger_catchup/normal_catchup.ml:83]: Call unnamed parameter last
```suggestion
      ~slot_tx_end ~slot_chain_end enveloped_initially_validated_transition
```

**Comment by joaosreis** [comment]
!ci-build-me

File: src/app/test_executive/slot_end_test.ml [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+  let fee = Currency.Fee.of_nanomina_int_exn 10_000_000
+
+  let amount = Currency.Amount.of_nanomina_int_exn 10_000_000
+
+  let tx_delay_ms = 5000
+
+  let run network t =
+    let open Malleable_error.Let_syntax in
+    let logger = Logger.create () in
+    let num_slots = slot_chain_end + 2 in
```
Comment (georgeee) [inline src/app/test_executive/slot_end_test.ml:80]: Please propagate this constant change to the base `compatble`-based branch

**Review by georgeee — APPROVED** [review]

**Comment by joaosreis** [comment]
!ci-build-me

**Review by SanabriaRusso — APPROVED** [review]

**Comment by mrmr1993** [comment]
!approved-for-mainnet

**Pull request merged**